### DISCUSSION TITLE
Is python supported in live code blocks?

### DISCUSSION THREAD

**Discussion by ghost** [quarto-dev/quarto-cli#8309] [Q&A] [answered]
### Description

The Docusaurus [docs for live code blocks](https://docusaurus.io/docs/markdown-features/code-blocks#interactive-code-editor) say:
> To use the plugin, create a code block with live attached to the language meta string.

The fact that it says generically "the language meta string" as implying that live code blocks should work for any supported language. And [in the Supported Languages section](https://docusaurus.io/docs/markdown-features/code-blocks#supported-languages) the docs say:
> By default, Docusaurus comes with a subset of [commonly used languages](https://github.com/FormidableLabs/prism-react-renderer/blob/master/packages/generate-prism-languages/index.ts#L9-L23).

Opening that link shows the supported languages are:
```
export const languagesToBundle = <const>[
  "markup",
  "jsx",
  "tsx",
  "swift",
  "kotlin",
  "objectivec",
  "js-extras",
  "reason",
  "rust",
  "graphql",
  "yaml",
  "go",
  "cpp",
  "markdown",
  "python",
]
```

Python is on that list (at the bottom). And yet I can't get python to work in a live code block. (I did get it to work with the jsx example in the docs, so I know I have the live code blocks plugin configured correctly). For example, when I try:
```
```python live
def my_function():
  print("Hello from a function")

my_function()
```

I just get an error:
```
SyntaxError: Unexpected token (1:12)
1 : return (def my_function():
```

So now I'm thinking python is not supported? Can anyone clarify? 

Thanks!

**Comment by mcanouil** [comment, marked as answer]
This is a Docusaurus feature which has nothing to do with Quarto.
Are you sure you are asking in the right place? (Docusaurus support: <https://docusaurus.io/community/support>)

The error message you show is from where? (Definitely not from Quarto)

Could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

![A placeholder image](https://placehold.co/600x400.png)

The end.
````
`````

**Reply by mcanouil** [reply]
To keep the code cell untouched, you need to tell it's raw:

`````qmd
---
format: docusaurus-md
---

````{=markdown}
```python live
def my_function():
  print("Hello from a function")

my_function()
```
````
`````

@cscheid Is there a bug here? I would have expected Quarto/Pandoc would not touch a code cell without brackets.

**Reply by cderv** [reply]
@mcanouil the syntax with `` ```python live `` is not a valid Quarto / Pandoc Markdown syntax for code block. So it will be parsed as usual Markdown. 

See how it is parsed by Pandoc 
```` 
> pandoc -t native
```python live
def my_function():
  print("Hello from a function")

my_function()
```

[ Para
    [ Str "```python"
    , Space
    , Str "live"
    , SoftBreak
    , Str "def"
    , Space
    , Str "my_function():"
    , SoftBreak
    , Str "print("
    , Quoted
        DoubleQuote
        [ Str "Hello"
        , Space
        , Str "from"
        , Space
        , Str "a"
        , Space
        , Str "function"
        ]
    , Str ")"
    ]
, Para [ Str "my_function()" , SoftBreak , Str "```" ]
]
````

Adding the raw block syntax makes sure it is untouched 
`````powershell
> pandoc -t native
````{=markdown}
```python live
def my_function():
  print("Hello from a function")

my_function()
```
````

[ RawBlock
    (Format "markdown")
    "```python live\ndef my_function():\n  print(\"Hello from a function\")\n\nmy_function()\n```"
]
`````

**Comment by ghost** [comment]
@mcanouil My apologies! I meant to post this on the Docusaurus GitHub. I had too many tabs open and somehow posted it here instead. Whoops — sorry about that
